docs/site/migration/components/project-layout.md

```diff
@@ -0,0 +1,131 @@
+---
+lang: en
+title: 'Migrating component project layout'
+keywords: LoopBack 4, LoopBack 3, Migration, Extensions, Components
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/migration-extensions-project-layout.html
+---
+
+{% include tip.html content="
+Missing instructions for your LoopBack 3 use case? Please report a [Migration docs issue](https://github.com/strongloop/loopback-next/issues/new?labels=question,Migration,Docs&template=Migration_docs.md) on GitHub to let us know.
+" %}
+
+The first step on the component migration journey is to reorganize the project
+files, change component's entry point and documentation for adding the component
+to the target application.
+
+## LoopBack 3 component layout & mounting
+
+A LB3 component is implemented as a function accepting the target app object and
+the configuration options.
+
+A minimal component consists of a single `index.js` file with the following
+content:
+
+```ts
+module.exports = function initializeComponent(loopbackApplication, options) {
+  // implementation
+};
+```
+
+A component is typically added to a LB3 application by creating a new entry in
+`server/component-config.json`, see
+[LoopBack components](https://loopback.io/doc/en/lb3/LoopBack-components.html).
+
+For example:
+
+```json
+{
+  "loopback-explorer": {
+    "mountPath": "/explorer"
+  }
+}
+```
+
+This allows the component to receive configuration. App developers can provide
+environment-specific configuration by using `component-config.{env}.json` files.
+
+## LoopBack 4 layout
+
+As explained in [Creating components](../../Creating-Components.md) and
+[Using components](../../Using-components.md), a typical LoopBack 4 component is
+an npm package exporting a Component class.
+
```

Comment:
We should recommend `lb4 extension` command to scaffold a package to expose components.

Reply:
I am already recommending `lb4 extension` elsewhere, see https://github.com/strongloop/loopback-next/pull/5262/files#diff-d72cf51faca1367bfb582f0995256991R111